### PR TITLE
Update vatlayer to use prices 1.0.0 and follow current code style and conventions

### PR DESCRIPTION
**Pull request by rafalp**
This PR updates library to work with types introduced in latest `prices` as well as does general maintenance on our codebase fixing code style, updating travis configs, setup.py, readme, test suite, repo setup, etc. ect.

* [x] Bump to py3k only
* [x] Clean tests
* [x] Test connectivity with vatlayer
* [x] Test lib against Django 1.11 and Django 2.0
* [x] Make `get_tax_rate` and `get_tax` work with passed down rates list
* [x] Cleanup deps
* [x] Update readme

For now I'm bumping version to `1.0.0b1`, as changes I've made are yet to be tested in other project.

### CODE REVIEW

**Comment by rafalp** [comment]
One more thing to add: Make `get_tax_rate_for_country` and `get_tax_for_country` work with passed down rates list, so we could those one time and pass around like how discounts at Saleor are working now.

**Comment by patrys** [comment]
When you say "passed down rate list" do you mean various rates for a single country or all rates for all countries? Just checking as the latter should not be necessary.

**Comment by rafalp** [comment]
Nope, was thinking about `vatlayers` middleware that would set `request.vatlayers` for valid country. So we are talking dict like such:

```
{
    "standard_rate":17,
    "reduced_rates": {  
        "wine": 14,
        "domestic fuel" :14,
        "advertising": 14,
        "bikes": 8,
        "domestic services": 8,
        "books": 3,
        "foodstuffs": 3,
        "pharmaceuticals": 3,
        "medical": 3,
        "passenger transport": 3,
        "newspapers": 3,
        "admission to cultural events": 3,
        "admission to sporting events": 3,
        "admission to entertainment events": 3,
        "hotels": 3,
        "restaurants": 3,
        "e-books": 3
    }
}
```

This would be passed to `*, standard_rate=None, reduced_rates=None` on `get_tax_for_country`. But I'll also need a way to opt-out when country is not using VAT, hitting DB @ `Vat.objects.get` for every price because `standard_rate is None` in your country will suck.

**Comment by rafalp** [comment]
Another pass on library API to make it better performer:

- Tax rates are read from cache and, if cache fails, read from DB.
- Tax rates are loaded and cached per-country, so users load exactly the amount of data that is needed for their use case.
- Tax rates are read once and passed to functions returning valid tax rate or tax function.

~~`get_tax_for_rate` will create new `Decimal` and closure with that decimal calling `prices.flat_tax`. Not careful dev may easily eat away some memory creating those functions all over the place, but others may pick their own strategy for keeping and reusing those functions, like how we are doing it with discounts in Saleor.~~ This is not different from Saleor using `functools.partial` for discounts.

File: django_prices_vatlayer/utils.py [hunk 1]
```diff
@@ -57,29 +62,47 @@ def create_objects_from_json(json_data):
     for code, value in rates.items():
         VAT.objects.update_or_create(
             country_code=code, defaults={'data': value})
+        get_tax_rates_for_country(code, force_refresh=True)
```
Comment (patrys) [inline django_prices_vatlayer/utils.py]: Wouldn't it be easier if `update_or_create` returned a structure that could be passed directly to `cache.set`?

**Comment by patrys** [comment]
By the way: it seems that the country code is not an indexed field.

**Comment by rafalp** [comment]
@patrys thanks, I've implemented those to lib's code.

**Pull request merged**